rails/README.md

```diff
@@ -34,10 +34,52 @@ A guide for building great Rails apps.
 
 ## Migrations
 
+Any time there is need to change the schema or data stored in the application
+database with a given group of work, write a [strong migration].
 - Don't change a migration after it has been merged into `main` if the desired
   change can be solved with another migration.
+- Schema migrations and data migrations should be kept separate.
+- We use the `strong_migration` gem, along with a 3 phase approach for any
+  change where we are renaming or moving columns around. These phases should
+  happen over 3 different releases.
+  - Phase 1:
+    - _Schema Migration_ - Create new column in the appropriate table.
+    - _Code and/or Schema Migration_ - Write the same data to both old and new
+      places.
+    - _Data Migration_ - Create a backfill so both columns have all the data.
+  - Phase 2:
+    - _Code_ - Move reads from old place to the new place.
+  - Phase 3:
+    - _Code and/or Schema Migration_ - Stop writing to old place.
+    - _Schema Migration_ - Remove old column.
+
+### Schema Migrations
+
+- These are standard rails migrations, they are stored in `db/migrate`.
+- No data should be migrated in these.
+- Since there are no data, there should be no references to models.
 - If there are default values, set them in migrations.
-- Use SQL, not `ActiveRecord` models, in migrations.
 - [Add foreign key constraints] in migrations.
+- Add appropriate indexes.
+
+### Data Migrations
+
+- Use SQL, not `ActiveRecord` models, in data migrations.
```

Comment:
We should mention the generator for data migrate we have: `rails g data_migration BackfillFoo`. This will nudge engineers towards using the `DataChange` tool we wrote, which in turn requires raw SQL.

We should also mention that the generator nudges us to write _reversible_ back fills, and we should try to do so when we can.